chiasim/ledger/ledger_api.py

```diff
@@ -163,6 +131,41 @@ def __init__(self, block_tip: HeaderHash, storage):
         r = await self._storage.unspent_for_coin_name(coin_name)
         return r
 
+    async def new_chain_view(self, chain_view, spend_bundles, coinbase_puzzle_hash,
+                             fees_puzzle_hash, storage, unspent_db):
+        block_number = chain_view.tip_index + 1
+
+        REWARD = int(1e9)
+        timestamp = uint64(self._now / 1000)
+
+        pool_public_key = get_pool_public_key()
+        plot_public_key = get_plot_public_key()
+
+        pos = ProofOfSpace(pool_public_key, plot_public_key)
+        coinbase_coin, coinbase_signature = create_coinbase_coin_and_signature(
+            block_number, coinbase_puzzle_hash, REWARD, pool_public_key)
+
+        spend_bundle = SpendBundle.aggregate(spend_bundles)
+
+        header, body = farm_new_block(
+            chain_view.tip_hash, chain_view.tip_signature, block_number, pos,
+            spend_bundle, coinbase_coin, coinbase_signature, fees_puzzle_hash,
+            timestamp)
+
+        header_signature = sign_header(header, plot_public_key)
+
+        [await storage.add_preimage(bytes(_)) for _ in (header, body)]
+
+        chain_view = await chain_view.augment_chain_view(
+            header, header_signature, storage, unspent_db, REWARD, self._now)
+
+        return chain_view
+
+    @api_request(ms=lambda x: int.from_bytes(x, 'big'))
```

Comment:
If you use `uint64` here, you don't need this `int.from_bytes` and `int_to_bytes` stuff (use `uint64.from_bytes`).

Reply:
updated